File: auto-docs/executables/julia/2d_histogram_scatter.jl
using Plotly
Plotly.signin("TestBot", "r1neazxo9w")
x0 = randn(100)/5. + 0.5
y0 = randn(100)/5. + 0.5
x1 = rand(50)
y1 = rand(50) + 1.0


x = [x0; x1]
y = [y0; y1]

trace1 = [
  "x" => x0, 
  "y" => y0, 
  "mode" => "markers", 
  "marker" => [
    "symbol" => "circle", 
    "opacity" => 0.7
  ], 
  "type" => "scatter"
]
trace2 = [
  "x" => x1, 
  "y" => y1, 
  "mode" => "markers", 
  "marker" => [
    "symbol" => "square", 
    "opacity" => 0.7
  ], 
  "type" => "scatter"
]
trace3 = [
  "x" => x, 
  "y" => y, 
  "type" => "histogram2d"
]
data = [trace1, trace2, trace3]
response = Plotly.plot(data, ["filename" => "2d-histogram-scatter", "fileopt" => "overwrite"])
plot_url = response["url"]

File: auto-docs/executables/python/2d_histogram_scatter.py
import numpy as np


x0 = np.random.randn(100)/5. + 0.5  # 5. enforces float division
y0 = np.random.randn(100)/5. + 0.5
x1 = np.random.rand(50)
y1 = np.random.rand(50) + 1.0


x = np.concatenate([x0, x1])
y = np.concatenate([y0, y1])
import plotly.plotly as py
from plotly.graph_objs import *
py.sign_in('TestBot', 'r1neazxo9w')
trace1 = Scatter(
    x=x0,
    y=y0,
    mode='markers',
    marker=Marker(
        symbol='circle',
        opacity=0.7
    )
)
trace2 = Scatter(
    x=x1,
    y=y1,
    mode='markers',
    marker=Marker(
        symbol='square',
        opacity=0.7
    )
)
trace3 = Histogram2d(
    x=x,
    y=y
)
data = Data([trace1, trace2, trace3])
plot_url = py.plot(data, auto_open=False, filename='2d-histogram-scatter')